src/eachvariate.jl

```diff
@@ -11,6 +11,16 @@ function EachVariate{V}(x::AbstractArray{<:Real,M}) where {V,M}
     return EachVariate{V,typeof(x),typeof(ax),T,M-V}(x, ax)
 end
 
+function ChainRulesCore.rrule(::Type{EachVariate{V}}, x::AbstractArray{<:Real,M}) where {V,M}
+    y = EachVariate{V}(x)
+    function EachVariate_pullback(Δ)
+        # TODO: Should we also handle `Tangent{<:EachVariate}`?
+        Δ_out = reshape(reduce(vcat, map(vec, ChainRulesCore.unthunk(Δ))), size(x))
```

Comment:
Often it's better to keep closed over variables as minimal as possible. Also `mapreduce` should be faster.
```suggestion
    size_x = size(x)
    function EachVariate_pullback(Δ)
        # TODO: Should we also handle `Tangent{<:EachVariate}`?
        Δ_out = reshape(mapreduce(vec, vcat, ChainRulesCore.unthunk(Δ)), size_x)
```

Reply:
The closure-capture was something I was also thinking about changing, but had forgotten about `mapreduce` :sweat_smile: